File: Swift/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.swift
import Foundation

func solution(_ lottos:[Int], _ win_nums:[Int]) -> [Int] {
    var count = 0
    var answer: [Int]
    
    for win_num in win_nums{
        if lottos.contains(win_num) {
            count += 1
        }
    }
    
    let zeroCount = lottos.filter { $0 == 0 }.count 
    
    let maxRank = min(6, 7 - (count + zeroCount))
    
    let minRank = min(6, 7 - count)
    
    answer = [maxRank, minRank]
    
    return answer
}

File: Python3/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    count = 0
    answer = []
    
    for win_num in win_nums:
        if win_num == 0:
            continue
        if win_num in lottos :
            count += 1
            
    if count > 1:
        answer.append(7-count)
    else:
        answer.append(6)
    
    if lottos.count(0) + count > 1:
        answer.append(7-lottos.count(0)-count)
    else:
        answer.append(6)
        
    return sorted(answer)
    